framework/src/main/java/org/checkerframework/common/wholeprograminference/WholeProgramInferenceJavaParserStorage.java

```diff
@@ -0,0 +1,1076 @@
+                        ClassOrInterfaceAnnos typeWrapper = new ClassOrInterfaceAnnos();
+                        if (!classToAnnos.containsKey(className)) {
+                            classToAnnos.put(className, typeWrapper);
+                        }
+
+                        sourceAnnos.types.add(typeWrapper);
+                    }
+
+                    @Override
+                    public void processMethod(
+                            MethodTree javacTree, MethodDeclaration javaParserNode) {
+                        addCallableDeclaration(javacTree, javaParserNode);
+                    }
+
+                    @Override
+                    public void processMethod(
+                            MethodTree javacTree, ConstructorDeclaration javaParserNode) {
+                        addCallableDeclaration(javacTree, javaParserNode);
+                    }
+
+                    /**
+                     * Creates a wrapper around {@code javacTree} with the corresponding declaration
+                     * {@code javaParserNode} and stores it.
+                     *
+                     * @param javacTree javac tree for declaration to add
+                     * @param javaParserNode JavaParser node for the same class as {@code javacTree}
+                     */
+                    private void addCallableDeclaration(
+                            MethodTree javacTree, CallableDeclaration<?> javaParserNode) {
+                        ExecutableElement elt = TreeUtils.elementFromDeclaration(javacTree);
+                        String className = ElementUtils.getEnclosingClassName(elt);
+                        ClassOrInterfaceAnnos enclosingClass = classToAnnos.get(className);
+                        String executableName = JVMNames.getJVMMethodSignature(javacTree);
+                        if (!enclosingClass.callableDeclarations.containsKey(executableName)) {
+                            enclosingClass.callableDeclarations.put(
+                                    executableName, new CallableDeclarationAnnos(javaParserNode));
+                        }
+                    }
+
+                    @Override
+                    public void processVariable(
+                            VariableTree javacTree, VariableDeclarator javaParserNode) {
+                        // This seems to occur when javacTree is a local variable in the second
+                        // class located in a source file. If this check returns false, then the
+                        // below call to TreeUtils.elementFromDeclaration causes a crash.
+                        if (TreeUtils.elementFromTree(javacTree) == null) {
+                            return;
+                        }
+
+                        VariableElement elt = TreeUtils.elementFromDeclaration(javacTree);
+                        if (!elt.getKind().isField()) {
+                            return;
+                        }
+
```

Comment:
It does. There's a method `processClass` that takes an `EnumDeclaration` as its second argument. It's called `processClass` because javac stores the enum declaration as a `ClassTree`.

Reply:
We would only add enum constants if we did inference on them, which we're not doing for now.